src/nrepl/project-types.ts

```diff
@@ -473,7 +473,7 @@ async function cljCommandLine(connectSequence: ReplConnectSequence, cljsType: Cl
         ...serverPrinterDependencies
     };
     const useMiddleware = [...middleware, ...(cljsType ? cljsMiddleware[cljsType] : [])];
-    const aliasesOption = aliases.length > 0 ? `-A${aliases.join("")}` : '';
+    const aliasesOption = aliases.length > 0 ? `-M${aliases.join("")}` : '-M';
```

Comment:
Is there a reason we add `-M` also when no aliases are selected?

Reply:
It's required to run a main function with args, it seems. It's not just about using aliases, although that is an option with `-M`. See https://clojure.org/reference/deps_and_cli#_running_a_main_or_script.

Also, the warning points to that fact as well: `WARNING: Implicit use of clojure.main with options is deprecated, use -M`.

If you take a jack-in command that doesn't include aliases, for which you see that warning, and you add `-M`. I was a bit confused about that as well.